### PR TITLE
Connect API

### PR DESCRIPTION
**Pull request by initseis**
In this branch I:

- Used redux-thunk middleware
- Updated components to use middlewares
- Created an API module
- Added thunk middlewares to reducers 

imagen

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,29 +1,63 @@
...
     default:
       return state;
   }
 };
 
+export const getBooksThunk = () => function (dispatch) {
+  getBooksAPI().then((res) => {
+    dispatch(getBooks(Object.entries(res)));
+  });
+};
```
Comment (dasileker) [inline src/redux/books/books.js]: - [ ] This is an unnamed function, so please refactor this function

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,29 +1,63 @@
...
+      ]),
+    );
+  });
+};
+
+export const removeBooksThunk = (id) => function (dispatch) {
+  deleteBookAPI(id).then(() => {
+    dispatch(removeBook(id));
+  });
+};
```
Comment (dasileker) [inline src/redux/books/books.js]: - [ ] Also this two are unnamed functions, refactor this please.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @sevinchek** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mricanho — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @sevinchek  🔥 

You worked hard in all the instructions that the previous Code Reviewers give you, ES6 are the best practices out there, good use of arrow functions👍🏽 

awesome-job

Now you can merge this branch 🚀 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by initseis** [comment]
Thanks, guys.

**Pull request merged**